crates/core/src/host/module_host.rs

```diff
@@ -760,8 +755,19 @@ impl ModuleHost {
         let db = self.inner.replica_ctx().relational_db.clone();
         // scheduled reducer name not fetched yet, anyway this is only for logging purpose
         const REDUCER: &str = "scheduled_reducer";
+        let caller_identity = self.info.identity;
+        let caller_address = self.info.address;
         self.call(REDUCER, move |inst: &mut dyn ModuleInstance| {
-            let mut tx = db.begin_mut_tx(IsolationLevel::Serializable);
+            let mut tx = db.begin_mut_tx(
+                IsolationLevel::Serializable,
+                Workload::Reducer(ReducerContext {
```

Comment:
I add this here thinking is more correct

Reply:
This pr fails `test_tx_inputs_are_in_the_commitlog`. Is because:

```rust
 fn do_durability(durability: &dyn Durability<TxData = Txdata>, ctx: &ExecutionContext, tx_data: &TxData) {
   ....
        } else {
...
            debug_assert!(!matches!(
                ctx.reducer_context().map(|rcx| rcx.name.strip_prefix("__identity_")),
                Some(Some("connected__" | "disconnected__"))
            ));
        }
    }```

So what does this mean here? @gefjon

Reply:
I think the issue is
```rust
// scheduled reducer name not fetched yet, anyway this is only for logging purpose
const REDUCER: &str = "scheduled_reducer";
```

Notice that `REDUCER` isn't the name of the reducer being scheduled.

I think what you want is the result of `call_reducer_params()` which returns a
```rust
CallReducerParams{ reducer_id, .. }
```

And then `self.info.reducers_map.lookup_name(reducer_id)` will give you the name of the reducer.

Reply:
Note: This comment is not for the failing test but how construct  the context of `REDUCER`